packages/grid/x-data-grid/src/colDef/gridStringOperators.ts

```diff
@@ -5,14 +5,19 @@ import { GridFilterItem } from '../models/gridFilterItem';
 import { GridFilterOperator } from '../models/gridFilterOperator';
 import { GridFilterInputMultipleValue } from '../components/panel/filterPanel/GridFilterInputMultipleValue';
 import { convertLegacyOperators, tagInternalFilter } from './utils';
+import { removeDiacritics } from '../hooks/features/filter/gridFilterUtils';
 
 export const getGridStringQuickFilterFn = tagInternalFilter(
   (value: any): GridApplyQuickFilterV7 | null => {
     if (!value) {
       return null;
     }
     const filterRegex = new RegExp(escapeRegExp(value), 'i');
-    return (columnValue): boolean => {
+    return (_, row, column, apiRef): boolean => {
+      let columnValue = apiRef.current.getRowFormattedValue(row, column);
+      if (apiRef.current.ignoreDiacritics) {
+        columnValue = removeDiacritics(columnValue);
+      }
```

Comment:
This is a disadvantage compared to v6 filtering, where it was possible to overwrite the `cellParams.formattedValue`.